services/director-v2/tests/unit/with_dbs/test_modules_dynamic_sidecar_docker_service_specs.py

```diff
@@ -434,6 +433,16 @@ async def test_merge_dynamic_sidecar_specs_with_user_specific_specs(
         app_settings=minimal_app.state.settings,
     )
     assert dynamic_sidecar_spec
+    dynamic_sidecar_spec_dict = dynamic_sidecar_spec.dict()
+    expected_dynamic_sidecar_spec_dict = AioDockerServiceSpec.parse_obj(
+        expected_dynamic_sidecar_spec
+    ).dict()
+    # ensure some entries are sorted the same to prevent flakyness
+    for sorted_dict in [dynamic_sidecar_spec_dict, expected_dynamic_sidecar_spec_dict]:
+        for key in ["DY_SIDECAR_STATE_EXCLUDE", "DY_SIDECAR_STATE_PATHS"]:
+            sorted_dict["TaskTemplate"]["ContainerSpec"]["Env"][key] = sorted(
```

Comment:
FYI: if this is a list, you can also sort in place https://docs.python.org/3/tutorial/datastructures.html#more-on-lists

```python
sorted_dict["TaskTemplate"]["ContainerSpec"]["Env"][key].sort()
```